### PR TITLE
Changes to better align `create-virtual-hosts` step with metal nodes

### PR DESCRIPTION
**Pull request by comay**
Update Bento box version for ubuntu-18.04 to 202005.21.0
Treat BCC_DEPLOY_NETWORK_VM the same as other boolean environment variables
Do the Ansible equivalent of a apt(8) dist upgrade as part of provisioning

Signed-off-by: David Comay <dcomay@bloomberg.net>

### CODE REVIEW

File: virtual/provisioners/finish.sh [hunk 1]
```diff
@@ -165,10 +176,20 @@ function configure_swap {
...
     KERNEL_VERSION_FILE=/vagrant/kernel-version
     if test -f "${KERNEL_VERSION_FILE}"; then
-        # shellcheck disable=SC1090,SC1091
+        # shellcheck disable=SC1090
         source "${KERNEL_VERSION_FILE}"
         apt-get install -y "linux-${KERNEL_VERSION}"
     fi
 }
 
+function enable_ipv6 {
```
Comment (justinjpacheco) [inline virtual/provisioners/finish.sh:185]:
```suggestion
function disable_ipv6 {
```

File: virtual/provisioners/finish.sh [hunk 2]
```diff
@@ -136,6 +138,15 @@ EOF
     apt-get update
 }
 
+# Taken from Ansible's safe upgrade logic for apt(8)
```
Comment (justinjpacheco) [inline virtual/provisioners/finish.sh:141]: do we want to `dist-upgrade` instead?

File: virtual/provisioners/finish.sh [hunk 3]
```diff
@@ -92,9 +92,11 @@ function main {
     create_operations_user
     configure_vagrant_user
     configure_apt
+    upgrade_apt
```
Comment (justinjpacheco) [inline virtual/provisioners/finish.sh:95]: should we combine these two apt tasks together?

File: virtual/provisioners/finish.sh (hunk 1 above)
Comment (comay) [reply virtual/provisioners/finish.sh:185]: LOL, I went back and forth on this. I tried to align the name of the function with the environment variable so if the latter is undefined, then we don't enable (well, proactively then disable) IPv6. If it is set, then we enable (well, leave it enabled) IPv6.

If you think it really is clearer the other way, I can change it but that was my rationale.

File: virtual/provisioners/finish.sh (hunk 2 above)
Comment (comay) [reply virtual/provisioners/finish.sh:141]: Currently the 8.1 upgrade playbook does the equivalent of a `apt-get upgrade` with the possible addition of new packages. Looking at the Ansible sources, I mimicked what that does.

**Pull request merged**